subsys/bluetooth/mesh/Kconfig.models

```diff
@@ -133,6 +133,7 @@ config BT_MESH_PONOFF_SRV
 	select BT_MESH_NRF_MODELS
 	select BT_MESH_DTT_SRV
 	select BT_MESH_ONOFF_SRV
+	select BT_MESH_MODEL_EXTENSIONS if BT_SETTINGS
```

Comment:
If settings are enabled, the extension API must be used, because power-up behavior is based on it. If settings aren't enabled, you may save some bytes for each server model in case you don't use subscriptions.

Reply:
I was actually thinking of doing this:
```
select BT_MESH_MODEL_EXTENSIONS if BT_SETTINGS
imply BT_MESH_MODEL_EXTENSIONS if !BT_SETTINGS
```
This will keep the extension API enabled even if settings were disabled. And then if user wants, it can disable the extension API at all. But were in doubts if it is really need.

Reply:
I'd prefer making an upstream PR that sets extensions to `default y`

Reply:
Hmm, this sort of creates the possibility of non-conformant implementation. If BT_SETTINGS is not enabled then power-up behavior will be broken and it is the implementer's responsibility. However, if extensions are not enabled, subscription behavior is broken for extended models. So, IMO, BT_MESH_MODEL_EXTENSIONS is mandatory for all extended models.

Reply:
@trond-snekvik not sure if is ok to make it enabled by default as mesh sample won't fit in RAM for microbit (https://github.com/zephyrproject-rtos/zephyr/pull/20103#issuecomment-545928504). Could be specifically disabled for microbit though.

Then I would force it for the extending models.

Reply:
`BT_MESH_MODEL_EXTENSION` is now forced for models which use the extension API.